src/main/java/com/ays/backend/user/service/impl/UserServiceImpl.java

```diff
@@ -86,5 +87,40 @@ public UserDTO getUserById(Long id) {
         ).orElseThrow(() -> new UserNotFoundException("User with id " + id + " not found"));
     }
 
+    @Override
+    public UserDTO deleteSoftUserById(Long id) {
+        User user = userRepository.findById(id)
+                .orElseThrow(() -> new UserNotFoundException("User with id " + id + " not found"));
+
+        user.setStatus(UserStatus.PASSIVE);
+
+        var deactivedUser =userRepository.save(user);
+
+        return userMapper.mapUsertoUserDTO(deactivedUser);
+    }
+
+    @Override
+    public UserDTO updateUserById(Long id, UpdateUserRequest updateUserRequest) {
+
+        User user = userRepository.findById(id)
+                .orElseThrow(() -> new UserNotFoundException("User with id " + id + " not found"));
+
+        user = User.builder()
```

Comment:
Please use MapStruct or something similar. And for the performance, please keep an eye for the usages.

Reply:
I just revised the code block through MapStruct.